File: vet.js
const fs = require('fs');
const path = require('path');

function analyzeHex(filePath) {
    if (!fs.existsSync(filePath)) {
        console.error(`Error: File ${filePath} not found.`);
        return;
    }

    console.log(`=== Hex Analysis Report (JS) for: ${filePath} ===`);

    let data;
    try {
        data = fs.readFileSync(filePath);
    } catch (e) {
        console.error(`Error reading file: ${e.message}`);
        return;
    }

    const totalBytes = data.length;
    let nullBytesCount = 0;
    let controlCharsCount = 0;
    let nonAsciiCount = 0;

    for (let i = 0; i < totalBytes; i++) {
        const byte = data[i];
        if (byte === 0) nullBytesCount++;
        if (byte < 32 && ![9, 10, 13].includes(byte)) controlCharsCount++;
        if (byte > 126) nonAsciiCount++;
    }

    // Display first 256 bytes hex
    console.log(`\n[First 256 bytes in Hex]:`);
    const slice = data.slice(0, 256);
    let hexRow = "";
    for (let i = 0; i < slice.length; i++) {
        hexRow += slice[i].toString(16).padStart(2, '0') + " ";
        if ((i + 1) % 16 === 0) {
            console.log(hexRow.trim());
            hexRow = "";
        }
    }
    if (hexRow) console.log(hexRow.trim());
    if (totalBytes > 256) console.log("... (truncated)");

    // Report
    console.log(`\n[Metrics]:`);
    console.log(`- Total Size: ${totalBytes} bytes`);
    console.log(`- Null Bytes (00): ${nullBytesCount}`);
    console.log(`- Control Characters: ${controlCharsCount} (Suspicious if high)`);
    console.log(`- Non-ASCII Bytes: ${nonAsciiCount} (Possible obfuscation)`);

    // Verdict
    console.log(`\n[Verdict]:`);
    const nonAsciiRatio = totalBytes > 0 ? nonAsciiCount / totalBytes : 0;

    if (nullBytesCount > 0 || controlCharsCount > 0) {
        console.log("🔴 HIGH RISK: Binary embedding or suspicious control characters detected.");
    } else if (nonAsciiRatio > 0.1) {
        console.log("🟡 MEDIUM RISK: High ratio of non-ASCII characters. Check for encoding attacks.");
    } else {
        console.log("🟢 LOW RISK: File appears to be clean text.");
    }
    
    console.log("=".repeat(40));
}

const args = process.argv.slice(2);
if (args.length < 1) {
    console.log("Usage: node vet.js <file_path>");
} else {
    analyzeHex(args[0]);
}

File: vet.py
import sys
import os

def analyze_hex(file_path):
    if not os.path.exists(file_path):
        print(f"Error: File {file_path} not found.")
        return

    print(f"=== Hex Analysis Report for: {file_path} ===")
    
    try:
        with open(file_path, 'rb') as f:
            data = f.read()
    except Exception as e:
        print(f"Error reading file: {e}")
        return

    total_bytes = len(data)
    non_ascii_count = 0
    control_chars_count = 0
    null_bytes_count = 0
    
    # Statistics
    for byte in data:
        if byte == 0:
            null_bytes_count += 1
        if byte < 32 and byte not in [9, 10, 13]: # Not TAB, LF, CR
            control_chars_count += 1
        if byte > 126:
            non_ascii_count += 1

    # Display first 256 bytes hex
    print(f"\n[First 256 bytes in Hex]:")
    hex_dump = " ".join(f"{b:02x}" for b in data[:256])
    for i in range(0, len(hex_dump), 48):
        print(hex_dump[i:i+48])
    
    if total_bytes > 256:
        print("... (truncated)")

    # Report
    print(f"\n[Metrics]:")
    print(f"- Total Size: {total_bytes} bytes")
    print(f"- Null Bytes (00): {null_bytes_count}")
    print(f"- Control Characters: {control_chars_count} (Suspicious if high)")
    print(f"- Non-ASCII Bytes: {non_ascii_count} (Possible obfuscation)")

    # Verdict
    print(f"\n[Verdict]:")
    if null_bytes_count > 0 or control_chars_count > 0:
        print("🔴 HIGH RISK: Binary embedding or suspicious control characters detected.")
    elif (non_ascii_count / total_bytes > 0.1) if total_bytes > 0 else 0:
        print("🟡 MEDIUM RISK: High ratio of non-ASCII characters. Check for encoding attacks.")
    else:
        print("🟢 LOW RISK: File appears to be clean text.")
    
    print("=" * 40)

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python vet.py <file_path>")
    else:
        analyze_hex(sys.argv[1])
